docs/csharp/language-reference/operators/boolean-logical-operators.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Boolean logical operators - C# Reference"
+description: "Learn about C# operators that perform logical negation, conjunction (AND), and inclusive and exclusive disjunction (OR) operations with Boolean operands."
+ms.date: 04/05/2019
+author: pkulikov
+f1_keywords: 
+  - "!_CSharpKeyword"
+  - "&_CSharpKeyword"
+  - "^_CSharpKeyword"
+  - "|_CSharpKeyword"
+  - "&&_CSharpKeyword"
+  - "||_CSharpKeyword"
+helpviewer_keywords: 
+  - "logical operators [C#]"
+  - "short-circuiting operators [C#]"
+  - "logical negation operator [C#]"
+  - "NOT operator [C#]"
+  - "! operator [C#]"
+  - "AND operator [C#]"
+  - "ampersand operator [C#]"
+  - "conjunction operator [C#]"
+  - "& operator [C#]"
+  - "exclusive OR operator [C#]"
+  - "XOR operator [C#]"
+  - "^ operator [C#]"
+  - "OR operator [C#]"
+  - "disjunction operator [C#]"
+  - "| operator [C#]"
+  - "conditional logical AND operator [C#]"
+  - "short-circuiting logical AND operator [C#]"
+  - "&& operator [C#]"
+  - "conditional logical OR operator [C#]"
+  - "short-circuiting logical OR operator [C#]"
+  - "|| operator [C#]"
+---
+# Boolean logical operators (C# Reference)
+
+The following operators perform logical operations with [bool](../keywords/bool.md) operands:
+
+- Unary [`!` (logical negation)](#logical-negation-operator-) operator.
+- Binary [`&` (logical AND)](#logical-and-operator-), [`|` (logical OR)](#logical-or-operator-), and [`^` (logical exclusive OR)](#logical-exclusive-or-operator-) operators. Those operators always evaluate both operands.
+- Binary [`&&` (conditional logical AND)](#conditional-logical-and-operator-) and [`||` (conditional logical OR)](#conditional-logical-or-operator-) operators. Those operators evaluate the second operand only if it's necessary.
+
+The `&` and `|` operators are also defined for `bool?` operands. For more information, see [The bool? type](../../programming-guide/nullable-types/using-nullable-types.md#the-bool-type) section of the [Using nullable types](../../programming-guide/nullable-types/using-nullable-types.md) article.
+
+For the operands of [integral](../keywords/integral-types-table.md) types, the `&`, `|`, and `^` operators perform bitwise logical operations.
+
+## Logical negation operator !
+
+The unary `!` operator computes logical negation of its operand. That is, it produces `true`, if the operand evaluates to `false`, and `false`, if the operand evaluates to `true`:
+
+[!code-csharp-interactive[logical negation](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Negation)]
+
+## Logical AND operator &amp;
+
+The `&` operator computes the logical AND of its operands. The result of `x & y` is `true` if both `x` and `y` evaluate to `true`. Otherwise, the result is `false`.
+
+The `&` operator evaluates both operands even if the first operand evaluates to `false`, so that the result must be `false` regardless of the value of the second operand. The following example demonstrates that behavior:
```

Comment:
As you introduce the concept of short-circuiting (or not), I think it's important to explain that `x` and `y` could be Boolean expressions (including method calls). In other words, the operands are not limited to variables.

Reply:
@BillWagner hm, I had a feeling I've explained it: the "evaluates" word means that an operand could be an expression (or not?). Also, at the example one of operands is a local function call, clearly not a variable. Do you suggest explicitly state in the article that operands might be expressions? I feel a bit odd here, because operands typically could be expressions and only exceptions to that rule are usually mentioned in the docs (for example, the `++` and `--` operators).

Reply:
@BillWagner I've updated the introductions to the examples for the `&`, `|`, `&&`, and `||` operators. Also, examples are updated: dotnet/samples#797

What do you think of a new text?